### PR TITLE
Manas/chart date parser fixes

### PR DESCRIPTION
**Pull request by man-shar**
add parsing to inferred dates. sort chart labels using the inferred date format when creating chart.

### CODE REVIEW

**Comment by man-shar** [comment]
we can possibly be smarter here to remove false positives and check actual data values too.

File: src/components/common/utils.jsx [hunk 1]
```diff
@@ -3,9 +3,11 @@ import dayjs from "dayjs";
 import customParseFormat from "dayjs/plugin/customParseFormat.js";
 import weekOfYear from "dayjs/plugin/weekOfYear.js";
 import advancedFormat from "dayjs/plugin/advancedFormat.js";
+import isoWeek from "dayjs/plugin/isoWeek";
 dayjs.extend(advancedFormat);
 dayjs.extend(weekOfYear);
 dayjs.extend(customParseFormat);
+dayjs.extend(isoWeek);
```
Comment (rishsriv) [inline src/components/common/utils.jsx:10]: TIL! Thanks for adding this :D

File: src/components/common/utils.jsx [hunk 2]
```diff
@@ -65,9 +67,88 @@ export function isDate(s, colName) {
     /year/gi.test(colName) ||
     /month/gi.test(colName) ||
     /date/gi.test(colName) ||
-    /week/gi.test(colName)
-  );
-  // return /^[0-9]{4}-[0-9]{2}-[0-9]{2}T[0-9]{2}:[0-9]{2}:[0-9]{2}$/gi.test(s);
+    /week/gi.test(colName);
+
+  // now to actually guess the date format
```
Comment (rishsriv) [inline src/components/common/utils.jsx:72]: Oh wow! This is super comprehensive. Thank you!

**Review by rishsriv — APPROVED** [review]
Very cool – TIL a lot of cool things about Dayjs

**Pull request merged**